File: logica-matematica/ejercicio-024/resoluciones/lucas-pajarito/inventario.js
const motos = [
    {
        modelo: "Yamaha MT-03",
        velocidadMaxima: 180
    },
    {
        modelo: "Honda CB190R",
        velocidadMaxima: 140
    },
    {
        modelo: "Kawasaki Z400",
        velocidadMaxima: 190
    },
    {
        modelo: "Suzuki Gixxer 250",
        velocidadMaxima: 150
    },
    {
        modelo: "KTM Duke 390",
        velocidadMaxima: 170
    }
];

function analizarInventario(motos) {

    console.log("=== INVENTARIO DE MOTOS ===");

    let velocidadMayor = motos[0].velocidadMaxima;
    let velocidadMenor = motos[0].velocidadMaxima;

    motos.forEach(moto => {

        if (moto.velocidadMaxima > velocidadMayor) {
            velocidadMayor = moto.velocidadMaxima;
        }

        if (moto.velocidadMaxima < velocidadMenor) {
            velocidadMenor = moto.velocidadMaxima;
        }

        console.log(`Modelo: ${moto.modelo}`);
        console.log(`Velocidad máxima: ${moto.velocidadMaxima} km/h`);
        console.log("------------------------------");

    });

    const rango = velocidadMayor - velocidadMenor;

    console.log(`Velocidad máxima registrada: ${velocidadMayor} km/h`);
    console.log(`Velocidad mínima registrada: ${velocidadMenor} km/h`);
    console.log(`Rango de velocidades: ${rango} km/h`);

}

analizarInventario(motos);

File: logica-general/ejercicio-024/resoluciones/lucas-pajarito/inventario.py
motos = [
    {
        "marca": "Yamaha",
        "modelo": "FZ25",
        "año": 2022,
        "estado_motor": "Excelente"
    },
    {
        "marca": "Honda",
        "modelo": "CB190R",
        "año": 2020,
        "estado_motor": "Bueno"
    },
    {
        "marca": "Suzuki",
        "modelo": "Gixxer 150",
        "año": 2019,
        "estado_motor": "Regular"
    },
    {
        "marca": "KTM",
        "modelo": "200 Duke",
        "año": 2018,
        "estado_motor": "Malo"
    },
    {
        "marca": "Bajaj",
        "modelo": "Pulsar NS200",
        "año": 2021,
        "estado_motor": "Bueno"
    }
]


def revisar_motor(motos):

    for moto in motos:

        if moto["estado_motor"] == "Excelente" or moto["estado_motor"] == "Bueno":
            print(f'{moto["marca"]} {moto["modelo"]}: Motor apto para seguir en uso.')
        else:
            print(f'{moto["marca"]} {moto["modelo"]}: Se recomienda cambiar el motor.')


revisar_motor(motos)